### PR TITLE
Introduce node IP address pool

### PR DESCRIPTION
**Pull request by hakonhall**
A host that is supposed to run containers has a non-empty IP pool: A set of
IPv4 and/or IPv6 addresses that can be assigned to containers.

This PR adds a list of hostnames to this pool.  The intent is that the
hostnames and IPs match through resolving, but resolution may not yet be
available (until DNS changes propagate).

For now, only a list of hostnames are specified.  We may want to specify
(hostname, IP address) pairs or (hostname, IPv4, IPv6) triplets later, and the
serialization format allows for that by storing the hsotnames in an array of
objects, the object having a "hostname" field.

However the REST API is kept simpler for now: it exposes and allows patching of
an array of strings of a "additionalHostnames" field.

### CODE REVIEW

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/persistence/NodeSerializer.java [hunk 1]
```diff
@@ -358,6 +371,16 @@ private ClusterMembership clusterMembershipFromSlime(Inspector object) {
...
 
+    private List<Address> addressesFromSlime(Inspector object) {
+        Inspector addressesField = object.field(containersKey);
+        if (addressesField.children() == 0)
+            return List.of();
+        List<Address> addresses = new ArrayList<>(addressesField.children());
+        addressesField.traverse((ArrayTraverser) (i, elem) ->
+                addresses.add(new Address(elem.field(containerHostnameKey).asString())));
+        return addresses;
+    }
```
Comment (freva) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/persistence/NodeSerializer.java:382]: Consider
```suggestion
    private List<Address> addressesFromSlime(Inspector object) {
        return SlimeUtils.entriesStream(object.field(containersKey))
                .map(elem -> new Address(elem.field(containerHostnameKey).asString()))
                .collect(Collectors.toList());
    }
```

File: node-repository/src/test/java/com/yahoo/vespa/hosted/provision/provisioning/AllocationSimulator.java [hunk 2]
```diff
@@ -11,8 +11,9 @@
 import com.yahoo.vespa.hosted.provision.NodeList;
 import com.yahoo.vespa.hosted.provision.node.Allocation;
 import com.yahoo.vespa.hosted.provision.node.Generation;
+import com.yahoo.vespa.hosted.provision.node.IP;
 
-import javax.swing.JFrame;
+import javax.swing.*;
```
Comment (freva) [inline node-repository/src/test/java/com/yahoo/vespa/hosted/provision/provisioning/AllocationSimulator.java:16]: Disable wildcard imports for `javax.swing.*`

**Review by freva — APPROVED** [review]

File: node-repository/src/test/java/com/yahoo/vespa/hosted/provision/provisioning/AllocationSimulator.java (hunk 2 above)
Comment (hakonhall) [reply node-repository/src/test/java/com/yahoo/vespa/hosted/provision/provisioning/AllocationSimulator.java]: huh, that was a weird exception in IntelliJ - removed and fixed.

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/persistence/NodeSerializer.java (hunk 1 above)
Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/persistence/NodeSerializer.java:382]: Sure

**Review by mpolden — APPROVED** [review]

**Pull request merged**